gx_suite_definitions/biomarkers.ipynb

```diff
@@ -0,0 +1,239 @@
+    "from expectations.expect_column_values_to_have_list_length import ExpectColumnValuesToHaveListLength\n",
+    "from expectations.expect_column_values_to_have_list_members import ExpectColumnValuesToHaveListMembers\n",
+    "from expectations.expect_column_values_to_have_list_members_of_type import ExpectColumnValuesToHaveListMembersOfType"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create Expectation Suite for Biomarkers Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "syn = synapseclient.Synapse()\n",
+    "syn.login()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "biomarkers_data_file = syn.get(\"syn63540269\").path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Validator Object on Data File"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_json(biomarkers_data_file)\n",
+    "nested_columns = ['points']\n",
+    "df = GreatExpectationsRunner.convert_nested_columns_to_json(df, nested_columns)\n",
+    "validator = context.sources.pandas_default.read_dataframe(df)\n",
+    "validator.expectation_suite_name = \"biomarkers\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Add Expectations to Validator Object For Each Column"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# points\n",
+    "validator.expect_column_values_to_be_of_type(\"points\", \"str\")\n",
+    "with open(\"../src/agoradatatools/great_expectations/gx/json_schemas/immunohisto/points.json\", \"r\") as file:\n",
+    "    points_schema = json.load(file)\n",
+    "validator.expect_column_values_to_match_json_schema(\"points\", json_schema=points_schema)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# model\n",
+    "validator.expect_column_values_to_be_of_type(\"model\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"model\")\n",
+    "# allows all alphanumeric characters, underscores, periods, and dashes\n",
+    "validator.expect_column_values_to_match_regex(\"type\", \"^[A-Za-z0-9\\s\\*_.-]+$\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# type\n",
+    "validator.expect_column_values_to_be_of_type(\"type\", \"str\")\n",
+    "validator.expect_column_values_to_not_be_null(\"type\")\n",
+    "# allows all alphanumeric characters, underscores, periods, and dashes\n",
+    "validator.expect_column_values_to_match_regex(\"type\", \"^[A-Za-z0-9\\s_.-]+$\")"
```

Comment:
Same question. Even better: do we have a limited list of types that can go in this field or is it a free-for-all?

Reply:
@JessterB do you know the answer to this question?

Reply:
It is not a free-for-all, each of the two datasets has a unique set of valid values.

Reply:
What are they and should we make a list of them instead of using regex?

Reply:
You should be able to use the unique set of values in the `type` column in the current files, as your list. There's code in one of the other GX suites that Brad wrote, that tells GX to compare values to a list of pre-defined values. Off the top of my head I don't remember which suite to point you to though. Biodomains probably has it?

Reply:
Yup, you are thinking of `expect_column_values_to_be_in_set` which is used in [this](https://github.com/Sage-Bionetworks/agora-data-tools/blob/dev/gx_suite_definitions/biodomain_info.ipynb) suite.